### DISCUSSION TITLE
Help trying to set up test environment testing/tutorials

### DISCUSSION THREAD

**Discussion by paurray** [templaza/astroid-framework#389] [Q&A] [answered]
Hello Astroid Community

- I am trying to install the test version of Astroid 3.
- Am running into some really basic problems in MAMP Pro
- Could really do with some help to proceed.

I believe that the below is preventing me from installing Astroid 3:

Dragging and dropping:

astroid-framework-3.zip

Here in Joomla/System/Install/Extensions/Upload on my MAMP Pro Virtual Host

Results in:

Installation of the library was successful.

But the Astroid Framework does not show up in Templates!

**********************************************************

a) In Joomla/System/Information/Warnings:

WarningPHP Upload Size bigger than POST size.
The value of the upload_max_filesize in the php.ini file is greater than the post_max_size variable. The post_max_size variable will take precedence and block requests larger than it. This is generally a server misconfiguration when trying to increase upload sizes. Please increase the value of post_max_size to be greater than the value of upload_max_filesize.

And

WarningSmall PHP maximum POST size.
This is the maximum amount of data that can be sent via POST to the server. This includes form submissions for articles, media (images, videos) and extensions. This value is less than 16MB which may impact on uploading large extensions. This is set in the php.ini under post_max_size.

**********************************************************

b) Here in Joomla/System/Information/System Information:

I see: PHP Version: 8.2.0

c) Here in Joomla/System/Information/System Information/PHP Settings:

I see: 

Upload Max Filesize	32M
Post Max Size	8M

d) Here in Joomla/System/Install/Extensions/Upload Package

I see Maximum upload size: ‎8.00 MB

**********************************************************

e) Here in
Applications/MAMP/bin/php/phpx.y.z/conf/php.ini

I can navigate to the folder php8.2.0

And in line 705 I find “post_max_size = 8M”

And change it to “post_max_size = 32M”

I stop/restart server and clear cache. No changes ripple through…

**********************************************************

I think I have some kind of fundamental mismatch going on. Maybe not mapping data bases. Or what ever.
Honestly I do not really know what I am doing.
I know that I was at least able to look as if what I was doing as I made all of these with MAMP Pro/Joomla/Astroid

https://www.youtube.com/@Astroid-Joomla-Seamlessly

**********************************************************

Can some one help me out?
I do not know maybe send me a correctly configured Joomla Install with MAMPRO? Or what ever I need to move forward?

Thanks for any help

Paul

**Comment by sonvnn** [comment]
Astroid package only 3MB. So I think it's not problem. May you record a video about your issue? It is easy for me to find solution.

Best Regards,
Sonny

**Comment by paurray** [comment]
Hi @sonvnn 
thank you for reaching out

I am using this as a rough road map but there are mistakes in it:
https://www.youtube.com/watch?v=GtwuHIefmAs
As in the Joomla install no longer goes in the Applications folder rater in
/Users/nameofcomputer/Sites/localhost/j

I am probably missing some really basic MAMP setting and would like to do a video of this working when it works for future reference… Also with the Astroid install for a test environment...

With MAMP installed:

01) Download latest Joomla from Joomla.org
02) Renamed the Joomla installation folder to “Astroid003MAMP“
03) And placed the folder here: /Users/name-of-computer/Sites/localhost/joomla/
04) In my browser I go here: http://localhost:8888/Astroid003MAMP/installation/index.php

05) 
Setup Site Name:
Astroid003test

06)
Enter the real name of your Super User. *:
paul
///
Set the username for your Super User account. *:
admin
///
Set the password for your Super User account. *:
admin1234567
///
Enter the email address of the website Super User. *:
paul@mydomain.com

NEXT
Either a username you created or a username provided by your host. *:
root
///
Either a password you created or a password provided by your host.:
root
///
Enter the database name. *:
Astroid003MAMP

Go here:

http://localhost:8888/Astroid003MAMP/installation/index.php?view=remove&layout=default
http://localhost:8888/Astroid003MAMP/administrator/

And do as shown in the video:

https://youtu.be/sYlA9kUlx_E

Thank you for your time

**Comment by Chacapamac** [comment]
Try:
Comment out output_buffering (Need to be Off for Joomla)
upload_max_filesize = 100M
post_max_size = 150M
max_execution_time = 1000
max_input_time = 1000
memory_limit = 1000M

**Comment by Chacapamac** [comment]
**Folder Mamp**
![My-Folder-Mamp](https://github.com/templaza/astroid-framework/assets/1907276/cdf2b1c4-3e3a-40ab-b616-c0928880820c)

**Adresse to reach websites** —> http://localhost/mainwebfolder/folderclient/

For sure, this is what I like to do but it give an idea of what you can do.

**Comment by paurray** [comment]
If any one has any ideas am all ears.

Did NOT CHANGED as I am guessing “off” is good!?!

Below the numbers at the front are the code line numbers

126 output_buffering
127 Default Value: Off
128 Development Value: 4096
129 Production Value: 4096

WAS:

859 upload_max_filesize = 32M

705 post_max_size = 32M

421 max_input_time = 60

432 memory_limit = 128M

NOW IS:

859 upload_max_filesize = 32M

705 post_max_size = 150M

421 max_input_time = 1000

432 memory_limit = 1000M

https://youtu.be/uICh-Z7lYn0

[php.ini.zip](https://github.com/templaza/astroid-framework/files/12646507/php.ini.zip)

**Comment by paurray** [comment]
In case this is useful to any one else…
This is what I missed:
https://support.mamp.info/knowledgebase.php?article=185

Have made changes suggested above and still not working for me though will do more digging next week...

<img width="1080" alt="ChangesWorkInJoomlaBackEnd" src="https://github.com/templaza/astroid-framework/assets/9163424/89e88437-a02f-4992-a026-923579fdcb04">

**Comment by Chacapamac** [comment, marked as answer]
@paurray  — **Do you still have the problem?**

![Screen Shot 2024-03-29 at 6 04 29 AM](https://github.com/templaza/astroid-framework/assets/1907276/02d1c4fa-ece2-421e-bc57-894e06655d1d)


**What can be tweaked up.**
;Comment Out —> output_buffering = 4096
max_execution_time = 300     ; Maximum execution time of each script, in seconds
max_input_time = 300	; Maximum amount of time each script may spend parsing request data
memory_limit = 256M      ; Maximum amount of memory a script may consume (32MB)
file_uploads	= On		; Whether to allow HTTP file uploads
;upload_tmp_dir	=	              ; temporary directory for HTTP uploaded files (will use system default if not specified (; in front of the line))
upload_max_filesize = 100M		    ; Maximum allowed size for uploaded files
and
post_max_size = 150M     ;Maximum size of POST data that PHP will accept. (Need to be bigger than upload - upload_max_filesize )

**Comment by paurray** [comment]
@Chacapamac thanks for getting back about this.
I solved the problem and if memory serves me correctly the same way of very similar to how you are suggesting. For me at least the link I posted above ended up being the key after a couple of false starts.

On a side note I have a nasty feeling that the old blue elephant has got lost in the Jungle:

https://x.com/mamp_en/status/1737114424400572698?s=20

Will miss the versioning etc but I have been using and am very happy with Herd for about 4 months now. Just do not see MAMP being updated EVER!

https://herd.laravel.com/
